lib/msf/core/modules/metadata/obj.rb

```diff
@@ -178,6 +181,8 @@ def init_from_hash(obj_hash)
     @post_auth          = obj_hash['post_auth']
     @default_credential = obj_hash['default_credential']
     @notes              = obj_hash['notes'].nil? ? {} : obj_hash['notes']
+    @needs_cleanup      = obj_hash['cleanup_required']
```

Comment:
Per: https://github.com/rapid7/metasploit-framework/pull/11903/files#diff-363256d380eb6bdc72aa7cd7f7a889b0R136 this should be `obj_hash['needs_cleanup']`